docs/vm-configuration.md

```diff
@@ -0,0 +1,265 @@
+have knowledge about the compute hosts which is required in order to make the
+correct choices.
+
+The public k8s virtual machine configuration will must allow users to fully
+specify the frontend virtual hardware, in order to enable a stable ABI to be
+attained.
+
+For the backend configuration, some parts be can automatically determined by
+kubevirt without requiring any user specification. For example, there is no need
+for the user to specify what VNC/SPICE port to use, or where to find the UEFI
+BIOS files. It is sufficient to know that VNC or UEFI must be enabled. Other
+areas of backend configuration will require the user to specify resources in an
+abstracted manner. For example, instead of providing a full file path for a disk,
+the tenant would provide the name of a PersistentVolumeClaim resource, or
+alternative k8s API resource. Further areas of backend configuration will
+require users to specify desired goals, which kubevirt then translates into
+specific configuration information. For example, the tenant may request exclusive
+use of CPUs, and kubevirt has to decide which host CPUs to bind virtual CPUs to,
+potentially impacting NUMA node placement of the guest as a result.
+
+Given a k8s API resource describing the virtual machine, it will be possible to
+generate a libvirt domain XML document from scratch. The particular API usage
+scenario, however, may affect what information is provided in the domain XML.
+The domain XML generated at the time a guest is started, will be the most fully
+specified XML in terms of both frontend and backend configuration. If domain XML
+is generated in isolation, however, while the frontend can be fully specified,
+the backend configuration may remain partially specified. For example, it would
+be possible to expand a PVC to a backend disk config, but without knowledge of
+a specific host to run the guest, the domain XML cannot contain information
+about CPU pinning.
+
+Going in the reverse direction is more difficult in general. The conversion
+from the high level k8s resource to the domain XML format is a lossy process.
+So given a domain XML document in isolation, it is not possible to generate a
+full k8s resource description of the virtual machine. At most it can contain
+frontend information. Translating the backend information in the domain XML, to
+the backend information required in the k8s resource is impractical or impossible
+in many cases. For example, there's no practical way to determine that a disk
+path on a particular host corresponds to a specific PVC. There is none the less
+a need for a conversion from domain XML to the k8s resource, in order to
+preserve the virtual hardware information for subsequent boots. Rather than
+trying to perform this conversion in isolation, however, it is more practical to
+take an initial k8s resource configuration (which contains all the high level
+backend information) and then augment the frontend configuration data with any
```

Comment:
The important thing is that you need *both* an original VM spec and the domain XML to update it with.